src/main/java/hudson/plugins/git/util/GitUtils.java

```diff
@@ -225,12 +242,12 @@ public static EnvVars getPollEnvironment(AbstractProject p, FilePath ws, Launche
                     }
                 }
             } else {
-                env = new EnvVars(System.getenv());
+                env = p.getEnvironment(workspaceToNode(ws), TaskListener.NULL);
             }
 
             p.getScm().buildEnvVars(b,env);
         } else {
-            env = new EnvVars(System.getenv());
+            env = p.getEnvironment(workspaceToNode(ws), listener);
```

Comment:
I don't think so. If using a remote workspace for polling, you assume a viable connection to slave. If this one fails then you won't get the expected result. I prefer polling to fail and wait for next cycle